### PR TITLE
Add paired and unpaired

### PR DESCRIPTION
**Pull request by jpallari**
Extension methods similar to `uncurry`. Converts parameters between a pair value and a pair of parameters.

### CODE REVIEW

File: src/main/kotlin/org/funktionale/currying/namespace.kt [hunk 1]
```diff
@@ -224,3 +224,11 @@ fun<P1, P2, P3, P4, P5, P6, P7, P8, P9, P10, P11, P12, P13, P14, P15, P16, P17,
 fun<P1, P2, P3, P4, P5, P6, P7, P8, P9, P10, P11, P12, P13, P14, P15, P16, P17, P18, P19, P20, P21, P22, R> ((P1) -> (P2) -> (P3) -> (P4) -> (P5) -> (P6) -> (P7) -> (P8) -> (P9) -> (P10) -> (P11) -> (P12) -> (P13) -> (P14) -> (P15) -> (P16) -> (P17) -> (P18) -> (P19) -> (P20) -> (P21) -> (P22) -> R).uncurried(): (P1, P2, P3, P4, P5, P6, P7, P8, P9, P10, P11, P12, P13, P14, P15, P16, P17, P18, P19, P20, P21, P22) -> R {
     return { p1: P1, p2: P2, p3: P3, p4: P4, p5: P5, p6: P6, p7: P7, p8: P8, p9: P9, p10: P10, p11: P11, p12: P12, p13: P13, p14: P14, p15: P15, p16: P16, p17: P17, p18: P18, p19: P19, p20: P20, p21: P21, p22: P22 -> this(p1)(p2)(p3)(p4)(p5)(p6)(p7)(p8)(p9)(p10)(p11)(p12)(p13)(p14)(p15)(p16)(p17)(p18)(p19)(p20)(p21)(p22) }
 }
+
+fun<P1, P2, R> Function2<P1, P2, R>.paired(): (Pair<P1, P2>) -> R {
+    return { pair: Pair<P1, P2> -> this(pair.component1(), pair.component2()) }
+}
+
+fun<P1, P2, R> Function1<Pair<P1, P2>, R>.unpaired(): (P1, P2) -> R {
+    return { p1: P1, p2: P2 -> this(Pair(p1, p2)) }
```
Comment (MarioAriasC) [inline src/main/kotlin/org/funktionale/currying/namespace.kt]: Is technically correct but Kotlin's idiomatic way to create `Pair<P1,P2>` is `p1 to p2` using the `to` extension function

File: src/test/kotlin/org/funktionale/currying/CurryingTest.kt [hunk 2]
```diff
@@ -46,4 +46,14 @@ class CurryingTest {
...
     }
+
+    @Test fun testPaired() {
+        val sum2ints = { x: Int, y: Int -> x + y }
+
+        val paired = sum2ints.paired()
+        val unpaired = paired.unpaired()
+
+        assertEquals(sum2ints(5, 9), paired(Pair(5, 9)))
+        assertEquals(paired(Pair(5, 9)), unpaired(5, 9))
```
Comment (MarioAriasC) [inline src/test/kotlin/org/funktionale/currying/CurryingTest.kt]: Same here and line 56

**Review by MarioAriasC — CHANGES_REQUESTED** [review]
It looks good but I'm not convinced about putting this on currying package. It should be in a different new package

File: src/main/kotlin/org/funktionale/currying/namespace.kt (hunk 1 above)
Comment (jpallari) [reply src/main/kotlin/org/funktionale/currying/namespace.kt]: Cool. I'll update it. 👍

**Review by MarioAriasC — APPROVED** [review]
👍

**Pull request merged**